logging/logging.go

```diff
@@ -58,10 +59,30 @@ func DisableLogging() {
 // EnableStructuredLogs replaces all logging functions with structured logging
 // variants.
 func EnableStructuredLogs() (func(), error) {
-	logger, err := zap.NewProduction()
-	if err != nil {
-		return func() {}, err
-	}
+	// Define level-handling logic.
+	highPriority := zap.LevelEnablerFunc(func(lvl zapcore.Level) bool {
+		return lvl >= zapcore.ErrorLevel
+	})
+	lowPriority := zap.LevelEnablerFunc(func(lvl zapcore.Level) bool {
+		return lvl < zapcore.ErrorLevel
+	})
+
+	// Lock console
+	consoleDebugging := zapcore.Lock(os.Stdout)
```

Comment:
We'll still want to send all logs to stderr unless a user has enabled the `log_debug_stdout` flag. The approach you had earlier passing the bool into `EnableStructuredLogs` still makes sense here. If the `log_debug_stdout` flag is set, we would use stdout on this line. Otherwise, it would be stderr.

Reply:
True, I've just fixed it.